src/test/java/com/test/combopack/packer/BruteForcePackagerTest.java

```diff
@@ -1,89 +1,37 @@
 
-import static org.junit.Assert.*;
-
+import org.junit.Before;
 import org.junit.Test;
 
 import com.pack.combopack.bean.Box;
 import com.pack.combopack.bean.Item;
-import com.pack.combopack.exception.PackagingException;
+import com.pack.combopack.bean.Packable;
 import com.pack.combopack.packer.BruteForcePackager;
-import com.pack.combopack.packer.KnapsackPackager;
-
-public class BruteForcePackagerTest {
-
-	KnapsackPackager packer = new BruteForcePackager();
-
-	@Test
-	public void pack() throws PackagingException {
-		for (Box<Item> box : availableBoxs) {
-			List<Item> items = packer.pack(getAvailableItems(box.getId()),
-					box.getMaxWeight());
-			assertTrue(getTotalWeight(items) < box.getMaxWeight());
-			box.setItems(items);
-			System.out.println(box);
-		}
-	}
-
-	@Test(expected = PackagingException.class)
-	public void packWithNULLItems() throws PackagingException {
-		packer.pack(null, 1.0d);
-
-	}
-
-	@Test(expected = PackagingException.class)
-	public void packWithEmptyItems() throws PackagingException {
-		packer.pack(new ArrayList<Item>(), 1.0d);
-
-	}
-
-	@Test(expected = PackagingException.class)
-	public void packWithInvalidWeight() throws PackagingException {
-		packer.pack(getAvailableItems(0), 0.0d);
-	}
-
-	@Test
-	public void packWithValidInput() throws PackagingException {
-		List<Item> items = packer.pack(getAvailableItems(0), 0.5d);
-		assertEquals("Total Size", items.size(), 1);
-		assertEquals(items.get(0).getValue(), 2.0, 0.01);
-	}
-
-	@Test(expected = PackagingException.class)
-	public void packWithMaxCapacity() throws PackagingException {
-		packer.pack(getAvailableItems(0), 101d);
 
-	}
+public class BruteForcePackagerTest extends AbstractKnapSackPackagerTest {
 
-	@Test(expected = PackagingException.class)
-	public void packWithMaxOverFullItems() throws PackagingException {
-		List<Item> overFillList = new ArrayList<Item>(getAvailableItems(3));
-		overFillList.addAll(getAvailableItems(3));
-		packer.pack(overFillList, 98d);
+    @Before
+    public void init() {
+        packer = new BruteForcePackager();
+    }
 
-	}
+    @Test
+    public void pack()  {
 
-	@Test(expected = PackagingException.class)
-	public void packWithItemWtViolation() throws PackagingException {
-		List<Item> list = new ArrayList<Item>(getAvailableItems(3));
-		list.get(0).setWeight(101d);
-		packer.pack(list, 100d);
+        System.out.println("\n <!-- Packaging with : " + packer.getClass().getSimpleName() + "-->");
 
-	}
+        for (Box<Item> box : availableBoxs) {
 
-	@Test(expected = PackagingException.class)
-	public void packWithItemValuetViolation() throws PackagingException {
-		List<Item> list = new ArrayList<Item>(getAvailableItems(3));
-		list.get(0).setValue(101);
-		packer.pack(list, 100d);
+            List<Packable> items = packer.pack(box);
+            assertTrue(packer.getTotalWeight(items) < box.getCapacity());
 
-	}
+            System.out.println("Id :" + box.getId() + " " + items);
+        }
+    }
```

Comment:
**Performance Issue**: The BruteForcePackager's pack method, as tested here, could lead to performance issues for large input sizes due to its brute force approach. This approach does not scale well with the number of items, potentially leading to exponential time complexity. <br> **Fix**: Consider implementing a more efficient algorithm, such as dynamic programming or a greedy approach, to handle larger datasets more effectively. This change could drastically improve the performance by reducing the time complexity. <br> **Code Suggestion**: 
 ```
 - packer = new BruteForcePackager();
 + packer = a new DynamicPackager(); // Or another more efficient implementation
 ```